test/test_torch.py

```diff
@@ -11040,6 +11040,77 @@ def test_exponential(self, device, dtype):
         with self.assertRaises(RuntimeError):
             torch.empty((1,), device=device, dtype=dtype).exponential_(-0.5)
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @dtypes(*(torch.testing.get_all_fp_dtypes(include_half=False) +
+              torch.testing.get_all_complex_dtypes()))
+    @dtypesIfCUDA(*(torch.testing.get_all_fp_dtypes(include_half=True) +
+                    torch.testing.get_all_complex_dtypes()))
+    def test_exp(self, device, dtype):
+        for v in (2, -2) + ((1j, 1 + 1j) if dtype.is_complex else ()):
+            if dtype == torch.bfloat16:
+                # Currently multiply a bfloat16 type with floating-point causes error:
+                #   RuntimeError: dtype != ScalarType::Undefined INTERNAL ASSERT FAILED at
+                #   "/pytorch/aten/src/ATen/native/TensorIterator.cpp":125, please report a bug to PyTorch.
+                # We skip bfloat16 for now, but we should fix it. https://github.com/pytorch/pytorch/issues/40580
+                if self.device_type == 'cpu' or self.device_type == 'cuda':
+                    with self.assertRaises(RuntimeError):
+                        torch.tensor(v, dtype=dtype, device=device) * torch.arange(18, device=device)
+                    return
+
+            a = torch.tensor(v, dtype=dtype, device=device) * torch.arange(18, device=device) / 3 * math.pi
+            a = a.to(dtype)
+            self.compare_with_numpy(torch.exp, np.exp, a)
+
+            if dtype.is_complex:
+                inf_real_zero_imag_in = torch.tensor(complex(float('inf'), 0), device=device, dtype=dtype)
+                inf_real_zero_imag_out = torch.exp(inf_real_zero_imag_in).item()
+                self.assertTrue(math.isinf(inf_real_zero_imag_out.real))
+                if device.startswith('cpu') and not IS_WINDOWS:  # Windows tests don't show some bugs consistently
+                    # This is incorrect. It should be zero. Need fix! https://github.com/pytorch/pytorch/issues/40590
+                    self.assertNotEqual(inf_real_zero_imag_out.imag, 0)
+                    # This is incorrect. They should equal. Need fix! https://github.com/pytorch/pytorch/issues/40590
+                    with self.assertRaises(AssertionError):
+                        self.compare_with_numpy(torch.exp, np.exp, inf_real_zero_imag_in)
+                elif not device.startswith('cpu'):
```

Comment:
You can just put `else:` here, right?

Reply:
I've dropped the `IS_WINDOWS` clause down one level, `else` should be sufficient here now.